float64 x
float64 y
float64 size_x
float64 size_y
